snuba/consumers/utils.py

```diff
@@ -0,0 +1,52 @@
+import time
+from logging import Logger
+
+from confluent_kafka import KafkaException
+from confluent_kafka.admin import AdminClient
+
+from snuba.utils.streams.configuration_builder import get_default_kafka_configuration
+from snuba.utils.streams.topics import Topic
+
+
+def get_partition_count_from_kafka_admin(topic: Topic, logger: Logger) -> int:
+
+    override_params = {
+        # Default is 60s, do we need that long?
+        "socket.timeout.ms": 2000,
+    }
+    attempts = 0
+    while True:
+        try:
+            logger.info("Attempting to connect to Kafka (attempt %d)...", attempts)
+            client = AdminClient(
+                get_default_kafka_configuration(
+                    topic=topic, override_params=override_params
+                )
+            )
+            cluster_metadata = client.list_topics(topic=topic.value, timeout=1.0)
+            break
+        except KafkaException as err:
+            logger.debug(
+                "Connection to Kafka failed (attempt %d)", attempts, exc_info=err
+            )
+            attempts += 1
+            # How many attempts is too many?
+            if attempts == 3:
+                raise
+            time.sleep(1)
+
+    try:
+        logger.info(f"Getting topic metadata for {topic.value}...")
+        topic_metadata = cluster_metadata.topics[topic.value]
+    except KeyError:
+        logger.info(f"No topic found for: {topic.value}")
```

Comment:
Are you sure you want to quietly swallow the exception here?

Reply:
Actually yeah that should probs raise an error, esp cause right now it will give an `UnboundLocalError: local variable 'topic_metadata' referenced before assignment`